state-chain/runtime/src/lib.rs

```diff
@@ -1262,6 +1262,23 @@ impl_runtime_apis! {
 		fn cf_free_balances(account_id: AccountId) -> Result<AssetMap<AssetAmount>, DispatchErrorWithMessage> {
 			LiquidityProvider::free_balances(&account_id).map_err(Into::into)
 		}
+		fn cf_lp_total_balances(account_id: AccountId) -> Result<AssetMap<AssetAmount>, DispatchErrorWithMessage> {
+			let free_balances = LiquidityProvider::free_balances(&account_id).map_err(Into::<DispatchErrorWithMessage>::into)?;
+			let open_order_balances = LiquidityPools::open_order_balances(&account_id);
+			let boost_pools_balances = {
+				let mut result = EthereumIngressEgress::boost_pool_balances(&account_id);
```

Comment:
My concern with this is that if we add another instance we're going to forget to add it here, but I don't think substrate makes this easy :/ . even using a macro, we'd have to remember to add the new instance to that macro... unless we processed the `construct_runtime!` macro to generate an enum with `EnumIter` on it for every instantiable pallet, which isn't a bad idea 🤔

Reply:
isn't this macro provided by substrate?

Reply:
We could write our own macro that processes their macro is what I mean, like a wrapper macro

Reply:
Oh yeah, definitely worth having a look at this, I'd create a ticket and work on that on a separate PR, what do you think? (is it worth having it in this PR, Idk how much work it could be)

Reply:
Yeah definitely a separate ticket